src/fiber/stack_pool.cr

```diff
@@ -0,0 +1,56 @@
+class Fiber
+  # :nodoc:
+  class StackPool
+    STACK_SIZE = 8 * 1024 * 1024
+
+    def initialize
+      @deque = Deque(Void*).new
+    end
+
+    # Removes and frees at most *count* stacks from the top of the pool,
+    # returning memory to the operating system.
+    def collect(count = lazy_size / 2)
+      count.times do
+        if stack = @deque.shift?
+          LibC.munmap(stack, STACK_SIZE)
+        else
+          return
+        end
+      end
+    end
+
+    # Removes a stack from the bottom of the pool, or allocates a new one.
+    def checkout
+      stack = @deque.pop? || allocate
+      {stack, stack + STACK_SIZE}
+    end
+
+    # Appends a stack to the bottom of the pool.
+    def release(stack)
```

Comment:
Isn't it bit of a weird name when you're actually _push_ing the stack into the pool?

Reply:
I used the namings from `DB::Pool`.